aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SdkSettings.kt

```diff
@@ -25,16 +25,14 @@ class SdkSettings private constructor(private val awsSdk: ObjectNode?) {
         get() =
             awsSdk?.getStringMember("defaultConfigPath")?.orNull()?.value.let { Paths.get(it) }
 
-    /** Path to the `sdk-endpoints.json` configuration */
-    val endpointsConfigPath: Path?
-        get() =
-            awsSdk?.getStringMember("endpointsConfigPath")?.orNull()?.value?.let { Paths.get(it) }
-
     /** Path to the `default-partitions.json` configuration */
     val partitionsConfigPath: Path?
         get() =
             awsSdk?.getStringMember("partitionsConfigPath")?.orNull()?.value?.let { Paths.get(it) }
 
+    val productionSdkBuild: Boolean
```

Comment:
As this is being set to true in non-production builds, perhaps we should call it, "requirePartitionsJson"?

Reply:
I guess "production" isn't the right word. I really mean "we're generating an AWS SDK" — I think we may want this in other places, e.g. we can remove the requirement for endpoint rules based on this flag instead of requiring customers to manually set the codegen flag.

Reply:
I think we were using "generateReadme" for that purpose previously.